### PR TITLE
String fixes

### PR DESCRIPTION
**Pull request by wesboyt**
*Issue #, if available:*
https://github.com/amzn/ion-js/issues/350
https://github.com/amzn/ion-js/issues/348
Removes equivs-string.ion and emptystrings.ion from the skiplists.
*Description of changes:*
Triple quote strings are currently corrupting the state of the parser when reading escaped terminator characters (single quotes). Values like: '''abc\\'''' are returning abc instead of abc'
This fix flags escaped characters in the parsing loop as non terminating.

It also makes sure characters read from read_escape_sequences() are valid  before adding them to the result string which fixes the escaped newline issues within strings.'

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/IonParserTextRaw.ts [hunk 1]
```diff
@@ -1072,13 +1074,9 @@ export class ParserTextRaw {
                         }
                     } else if (this.isLowSurrogate(ch)) {
                         throw new Error("unexpected low surrogate: " + ch);
-                    } else if (t === T_STRING3 && ch === CH_SQ) {
-                        if (this.verifyTriple(index)) {
+                    } else if (t === T_STRING3 && ch === CH_SQ && !bs && this.verifyTriple(index)) {
                             index = this._skip_triple_quote_gap(index, this._end, /*acceptComments*/ true);
-                        } else {
```
Comment (wesboyt) [inline src/IonParserTextRaw.ts]: this character check skips characters like escaped newlines which return negative values.

File: src/IonParserTextRaw.ts [hunk 2]
```diff
@@ -1072,13 +1074,9 @@ export class ParserTextRaw {
                         }
                     } else if (this.isLowSurrogate(ch)) {
                         throw new Error("unexpected low surrogate: " + ch);
-                    } else if (t === T_STRING3 && ch === CH_SQ) {
```
Comment (wesboyt) [inline src/IonParserTextRaw.ts]: the important check is !bs, the other alterations are just a simplification.

File: src/IonParserTextRaw.ts [hunk 3]
```diff
@@ -1049,10 +1049,12 @@ export class ParserTextRaw {
             case T_STRING2:
             case T_STRING3:
                 for (index = this._start; index < this._end; index++) {
+                    let bs = false;
```
Comment (wesboyt) [inline src/IonParserTextRaw.ts]: bs is set to false before each character is read.

File: src/IonParserTextRaw.ts [hunk 4]
```diff
@@ -1072,13 +1074,9 @@ export class ParserTextRaw {
...
                         throw new Error("unexpected low surrogate: " + ch);
-                    } else if (t === T_STRING3 && ch === CH_SQ) {
-                        if (this.verifyTriple(index)) {
+                    } else if (t === T_STRING3 && ch === CH_SQ && !bs && this.verifyTriple(index)) {
                             index = this._skip_triple_quote_gap(index, this._end, /*acceptComments*/ true);
-                        } else {
-                            s += String.fromCharCode(ch);
-                        }
-                    } else {
+                    } else if (ch >= 0) {
```
Comment (wesboyt) [inline src/IonParserTextRaw.ts]: this character check skips characters like escaped newlines which return negative values.

File: src/IonParserTextRaw.ts [hunk 5]
```diff
@@ -1072,13 +1074,9 @@ export class ParserTextRaw {
                         }
                     } else if (this.isLowSurrogate(ch)) {
                         throw new Error("unexpected low surrogate: " + ch);
-                    } else if (t === T_STRING3 && ch === CH_SQ) {
-                        if (this.verifyTriple(index)) {
+                    } else if (t === T_STRING3 && ch === CH_SQ && !bs && this.verifyTriple(index)) {
```
Comment (wesboyt) [inline src/IonParserTextRaw.ts]: the important check is !bs, the other alterations are just a simplification.

File: src/IonParserTextRaw.ts (hunk 3 above)
Comment (tgregg) [reply src/IonParserTextRaw.ts]: For readability, consider naming `bs` something like `isEscaped` (or, at least, `backslash`).

**Pull request merged**